src/Playwright/Transport/Converters/EvaluateArgumentValueConverter.cs

```diff
@@ -260,18 +260,25 @@ private static object ParseEvaluateResult(JsonElement result, Type t, JsonSerial
                     }
                 }
 
-                object objResult = Activator.CreateInstance(t);
-
-                foreach (var kv in keyValues)
+                try
                 {
-                    var serializerOptions = JsonExtensions.GetNewDefaultSerializerOptions();
+                    object objResult = Activator.CreateInstance(t);
 
-                    var property = t.GetProperties().FirstOrDefault(prop => string.Equals(prop.Name, kv.K, StringComparison.OrdinalIgnoreCase));
-                    serializerOptions.Converters.Add(GetNewConverter(property.PropertyType));
-                    property.SetValue(objResult, kv.V.ToObject(property.PropertyType, serializerOptions));
-                }
+                    foreach (var kv in keyValues)
+                    {
+                        var serializerOptions = JsonExtensions.GetNewDefaultSerializerOptions();
 
-                return objResult;
+                        var property = t.GetProperties().FirstOrDefault(prop => string.Equals(prop.Name, kv.K, StringComparison.OrdinalIgnoreCase));
+                        serializerOptions.Converters.Add(GetNewConverter(property.PropertyType));
+                        property.SetValue(objResult, kv.V.ToObject(property.PropertyType, serializerOptions));
+                    }
+
+                    return objResult;
+                }
+                catch (Exception)
```

Comment:
is this change related to the roll?

Reply:
Sort of. It's a consequence of a problem I was diagnosing, but it also makes sense to do it parallel to this.